### PR TITLE
PLT-108: Pretty instances for regression test types

### PR DESCRIPTION
**Pull request by zliu41**
This makes it possible to render and print test failures.

### CODE REVIEW

**Review by michaelpj — APPROVED** [review]

**Review by effectfully — COMMENTED** [review]
I don't know much about what's going on here. Superficially looks good.

File: plutus-ledger-api/testlib/PlutusLedgerApi/Test/EvaluationEvent.hs [hunk 1]
```diff
@@ -12,42 +14,84 @@ module PlutusLedgerApi.Test.EvaluationEvent (
...
     deriving anyclass (Serialise)
 
--- | Information about an on-chain script evaluation event, specifically the information needed
--- to evaluate the script, and the expected result.
+instance Pretty ScriptEvaluationData where
+    pretty ScriptEvaluationData{..} =
+        vsep
+            [ "protocol version:" <+> pretty dataProtocolVersion
+            , "budget: " <+> pretty dataBudget
+            , "script: " <+> pretty (Text.decodeLatin1 . Base64.encode $ BS.fromShort dataScript)
```
[the remark this answers is not in the document]
Comment (zliu41) [reply plutus-ledger-api/testlib/PlutusLedgerApi/Test/EvaluationEvent.hs:55]: This function is repeated multiple times in multiple repos :joy: 
It's usually for printing block hashes, but in this case it's Base64, not Base16, so it's different.

File: plutus-ledger-api/testlib/PlutusLedgerApi/Test/EvaluationEvent.hs [hunk 2]
```diff
@@ -12,42 +14,84 @@ module PlutusLedgerApi.Test.EvaluationEvent (
...
+            , "data: "
+                <+> pretty
```
[the remark this answers is not in the document]
Comment (zliu41) [reply plutus-ledger-api/testlib/PlutusLedgerApi/Test/EvaluationEvent.hs:57]: Right now `Pretty Data` isn't very pretty due to `show`ing ByteString directly, but as discussed improving `Pretty Data` is the better solution, so I'll update it here.

File: plutus-ledger-api/testlib/PlutusLedgerApi/Test/EvaluationEvent.hs [hunk 3]
```diff
@@ -71,6 +115,26 @@ data UnexpectedEvaluationResult
...
+
+instance Pretty UnexpectedEvaluationResult where
+    pretty = \case
+        UnexpectedEvaluationSuccess ev params budget ->
+            nest 2 $
+                vsep
+                    [ "UnexpectedEvaluationSuccess"
+                    , pretty ev
+                    , "Cost parameters:" <+> pretty params
+                    , "Budget spent:" <+> pretty budget
```
[the remark this answers is not in the document]
Comment (michaelpj) [reply plutus-ledger-api/testlib/PlutusLedgerApi/Test/EvaluationEvent.hs:123]: Ah, okay.

**Pull request merged**